frontend/src/io-managers/fonts.ts

```diff
@@ -58,19 +64,18 @@ export function createFontsState(editor: Editor) {
 					const files = new Map(font.variants.map((x) => [formatFontStyleName(x), font.files[x]]));
 					return { family, variants, files };
 				});
-				state.fontNames = result.map((value) => value.family);
 
 				resolve(result);
 			});
 	});
 
 	return {
-		state: readonly(state) as typeof state,
+		fontNames,
 		getFontStyles,
 		getFontFileUrl,
 	};
```

Comment:
I may have misdirected you about this being an I/O manager. The format for all I/O managers is that they must return from their constructor either nothing or a single destructor function. Not an object with several functions. I don't think `utility-functions` is the correct folder for this either, though, because those are all stateless single-purpose functions. Feel free to read `frontend/src/README.md` for a bit more detail on each of these types. Do you think this could maybe be restructured to fit the conventions for any of the others? I assume you can't make this stateless to work with `utility-functions`. Maybe the getter functions could be changed into entries for the `state` object and this could become a state provider again? Happy to discuss this further, I don't mean to misdirect you again.

Reply:
Why should they become entries in a state object? This would be additional complexity for no gain.

Reply:
I'm just trying to figure out how we can make it match the code style of the other files to avoid confusing inconsistency. My idea about `state` was that, perhaps instead of exposing these getter functions, we can instead just bind the current value to the `state` object so users can replace their getter function calls with reading the state values? I don't know if that is the right approach here or not, but just an idea. If you'd like to talk more about this on Discord, we can probably settle it out real quick, I just want to make sure we're using our frontend architecture in a not-inconsistent way.